content/automate/ManageIQ/Cloud/VM/Retirement/Email.class/__methods__/vm_retire_extend.rb

```diff
@@ -3,63 +3,82 @@
 # VM has a retires_on value and is not already retired
 #
 
-vm_retire_extend_days = nil
-vm_retire_extend_days ||= $evm.object['vm_retire_extend_days']
-raise "ERROR - vm_retire_extend_days not found!" if vm_retire_extend_days.nil?
-
-$evm.log("info", "Number of days to extend: <#{vm_retire_extend_days}>")
-
-obj = $evm.object("process")
-
-vm = obj["vm"] || $evm.root["vm"]
-vm_name = vm.name
-
-if vm.retires_on.nil? | vm.retires_on == ""
-  $evm.log("info", "VM '#{vm_name}' has no retirement date - extension bypassed")
-  exit MIQ_OK
-end
-
-if vm.retired
-  $evm.log("info", "VM '#{vm_name}' already marked as retired. vm.retires_on date: #{vm.retires_on}. No Action taken")
-  exit MIQ_OK
-end
-
-$evm.log("info", "VM: <#{vm_name}> current retirement date is #{vm.retires_on}")
-
-unless vm.retires_on.nil?
-  $evm.log("info", "Extending retirement <#{vm_retire_extend_days}> days for VM: <#{vm_name}>")
-
-  # Set new retirement date here
-  vm.extend_retires_on(vm_retire_extend_days, vm.retires_on)
-
-  $evm.log("info", "VM: <#{vm_name}> new retirement date is #{vm.retires_on}")
-
-  $evm.log("info", "Inspecting retirement: <#{vm.retirement.inspect}>")
-
-  evm_owner_id = vm.attributes['evm_owner_id']
-  owner = nil
-  owner = $evm.vmdb('user', evm_owner_id) unless evm_owner_id.nil?
-  $evm.log("info", "Inspecting VM Owner: #{owner.inspect}")
-
-  if owner
-    to = owner.email
-  else
-    to = $evm.object['to_email_address']
+module ManageIQ
+  module Automate
+    module Cloud
+      module VM
+        module Retirement
+          module Email
+            class VmRetireExtend
+              def initialize(handle = @evm)
+                @handle = handle
+              end
+
+              def main
+                @handle.log("info", "Starting vm_retire_extend")
+                vm = @handle.root['vm']
+                check_retire_extend(vm)
```

Comment:
@billfitzgerald0120 
We should check if the vm variable is set and raise an error if the vm doesn't exist

```
def vm
   raise "ERROR - vm object not passed in" unless @handle.root['vm']
   @handle.root['vm']
end
```

If we have the above method then in the main we can just call
```
def main
   check_retire_extend(vm)
end
```